core/arch/mangle_shared.c

```diff
@@ -61,8 +61,8 @@ callee_info_t default_callee_info;
 int
 get_clean_call_switch_stack_size(void)
 {
-#ifdef AARCH64
-    /* Stack size needs to be 16 byte aligned on ARM */
+#if defined(AARCH64) || defined(X64)
+    /* Stack size needs to be 16 byte aligned on ARM and x64. */
```

Comment:
I've added `|| MACOS`. Do you think it's worth adding a `NEEDS_16_BYTE_SP_ALIGNMENT` and use it a various places? To make it easier to keep the condition in sync. 

ARM/AArch32 needs 8 byte alignment, I think we should assert that in prepare_for_clean_call and cleanup_after_clean_call.